Rules/ReviewUnusedParameter.cs

```diff
@@ -51,6 +51,29 @@ public IEnumerable<DiagnosticRecord> AnalyzeScript(Ast ast, string fileName)
                     continue;
                 }
 
+                // bail out if $MyInvocation.BoundParameters used
+                if (variableCount.ContainsKey("MyInvocation"))
+                {
+                    bool isBound = false;
+                    IEnumerable<Ast> memExpAst = scriptBlockAst.FindAll(oneAst => oneAst is MemberExpressionAst, false);
+                    foreach (MemberExpressionAst memberExpressionAst in memExpAst)
+                    {                        
+                        if (memberExpressionAst.Expression is VariableExpressionAst &&
+                            ((VariableExpressionAst)memberExpressionAst.Expression).VariablePath.UserPath == "MyInvocation" &&
+                            memberExpressionAst.Member is StringConstantExpressionAst &&
+                            ((StringConstantExpressionAst)memberExpressionAst.Member).Value == "BoundParameters")
+                        {
+                            isBound = true;
+                            break;
+                        }                        
+                    }
+
+                    if (isBound)
+                    {
+                        continue;
+                    }
+                }
+

```

Comment:
What about `$PSCmdlet.MyInvocation.BoundParameters`?

Reply:
Good catch, let me enhance this condition to handle this as well.

Reply:
I have tweaked this for `$PSCmdlet.MyInvocation.BoundParameters` as well, do you think it can be optimized further or is this okay

Reply:
I have different approach, will implement it and let you know